packages/typespec-azure-portal-core/src/decorators.ts

```diff
@@ -0,0 +1,253 @@
+import { ArmResourceKind, getArmResourceKind } from "@azure-tools/typespec-azure-resource-manager";
+import {
+  StringLiteral,
+  getDirectoryPath,
+  getSourceLocation,
+  normalizePath,
+  resolvePath,
+  validateDecoratorUniqueOnNode,
+  type DecoratorContext,
+  type Model,
+  type ModelProperty,
+  type Program,
+  type Type,
+} from "@typespec/compiler";
+import fs from "fs";
+import { PortalCoreKeys } from "./keys.js";
+import { reportDiagnostic } from "./lib.js";
+import { AboutOptions, BrowseOptions, marketplaceOfferOptions } from "./types.js";
+
+/**
+ * This is a Browse decorator which will be use to put more info on the browse view.
+ * @param target The model that is being decorated.
+ * @param options BrowseOptions of the property.
+ */
+export function $browse(context: DecoratorContext, target: Model, options: BrowseOptions) {
+  const { program } = context;
+  validateDecoratorUniqueOnNode(context, target, $browse);
+  isARMResource(program, target, "browse");
+  if (options && (options as Model).properties) {
+    const query = (options as Model).properties.get("argQuery");
+    if (query && query.type.kind === "Model") {
+      const sourceLocation = getSourceLocation(target);
+      const dirPath =
+        sourceLocation && sourceLocation.file.path && getDirectoryPath(sourceLocation.file.path);
+      let argQueryPath = query.type && (query.type as Model).properties.get("filePath");
+      const argQueryPathValue =
+        argQueryPath && argQueryPath.type && (argQueryPath.type as StringLiteral).value;
+      let filePath = resolvePath(dirPath, argQueryPathValue); //if given path is fullpath, it will return the fullPath
+      if (filePath && argQueryPath && argQueryPathValue) {
+        filePath = normalizePath(filePath);
+        if (!fs.existsSync(filePath)) {
```

Comment:
Need to use `ctx.program.host.stat` instead of `fs.existsSync` here for correctness.

Reply:
will do! :) thank you so much!